airbyte-integrations/connectors/source-google-search-console/source_google_search_console/streams.py

```diff
@@ -0,0 +1,255 @@
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return "date"
+
+    @property
+    def http_method(self) -> str:
+        return "POST"
+
+    def next_page_token(self, response: requests.Response) -> Optional[int]:
+        if len(response.json().get(self.data_field, [])) == ROW_LIMIT:
+            self.start_row += ROW_LIMIT
+            return self.start_row
+
+        return None
+
+    def request_headers(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Mapping[str, Any]:
+        headers = {"Content-Type": "application/json"}
+        return headers
+
+    def request_body_json(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Optional[Union[Dict[str, Any], str]]:
+
+        start_date = self._start_date
+        if start_date and stream_state:
+            if stream_state.get(self.cursor_field):
+                start_date = max(
+                    pendulum.parse(stream_state[self.cursor_field]),
+                    pendulum.parse(start_date),
+                ).to_date_string()
+
+        data = {
+            "startDate": start_date,
+            "endDate": stream_state.get("end_date") or pendulum.now().to_date_string(),
+            "dimensions": ["date", self.value_field],
+            "searchType": "web",
+            "aggregationType": "auto",
+            "startRow": next_page_token or 0,
+            "rowLimit": ROW_LIMIT,
+        }
+        return data
+
+    def parse_response(
+        self, response: requests.Response, **kwargs
+    ) -> Iterable[Mapping]:
+        records = response.json().get(self.data_field) or []
+        for record in sorted(records, key=lambda x: x["keys"][0]):
+            record["date"] = record["keys"][0]
+            record[self.value_field] = record.pop("keys")[1]
+            yield record
+
+    def get_updated_state(
+        self,
+        current_stream_state: MutableMapping[str, Any],
+        latest_record: Mapping[str, Any],
+    ) -> Mapping[str, Any]:
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {
+                self.cursor_field: max(
+                    latest_benchmark, current_stream_state[self.cursor_field]
+                )
+            }
+        return {self.cursor_field: latest_benchmark}
+
+
+class SearchAnalyticsByCountry(SearchAnalytics):
+    value_field = "country"
+
+
+class SearchAnalyticsByDevice(SearchAnalytics):
+    value_field = "device"
+
+
+class SearchAnalyticsByPage(SearchAnalytics):
+    value_field = "page"
+
+
+class SearchAnalyticsByQuery(SearchAnalytics):
+    value_field = "query"
+
+
+class SearchAnalyticsByDate(SearchAnalytics):
+    def request_body_json(
```

Comment:
Did you override it just because it has only one dimension "date"?

Reply:
Yep

Reply:
I believe, there is better solution.